src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -833,6 +836,15 @@ FixedwingPositionControl::control_position(const Vector2f &curr_pos, const Vecto
 
 			float alt_sp = pos_sp_curr.alt;
 
+			if (pos_sp_next.type == position_setpoint_s::SETPOINT_TYPE_LAND && pos_sp_next.valid) {
```

Comment:
Keep in mind that you might be outside of an actual loiter while still in the SETPOINT_TYPE_LOITER handling. 

What do you think about waiting to enter the actual loiter (requiring `_l1_control.circle_mode()` here).

Reply:
Changed.